README.md

```diff
@@ -1 +1,404 @@
+
+There are **two** major serialized data formats:  
+
+* **JSON** stands for "JavaScript Object Notation", and has become a universal standard for serializing native data structures for transmission. It is light-weight, easy to read, and quick to parse.
+
+```json
+{
+  "users": [
+    {"name": "Bob", "id": 23},
+    {"name": "Tim", "id": 72}
+  ]
+}
+```
+> Remember, JSON is a serialized format. While it may look like an object, it needs to be parsed so we can interact with it as a true Javascript object.
+
+* **XML** stands for "eXtensible Markup Language", and is the granddaddy of serialized data formats (itself based on HTML). XML is fat, ugly, and cumbersome to parse. However, it remains a major format due to its legacy usage across the web. You'll probably always favor using a JSON API, if available.
+
+```
+<users>
+  <user id="23">
+    <name><![CDATA[Bob]]></name>
+  </user>
+  <user id="72">
+    <name><![CDATA[Tim]]></name>
+  </user>
+</users>
+```
+
+**Many APIs publish data in multiple formats, for example...**
+
+* [http://dev.markitondemand.com/Api/Quote/json?symbol=AAPL](http://dev.markitondemand.com/Api/Quote/json?symbol=AAPL)
+* [http://dev.markitondemand.com/Api/Quote/xml?symbol=AAPL](http://dev.markitondemand.com/Api/Quote/xml?symbol=AAPL)
+
+## Where Do We Find APIs? (5 mins)
+
+APIs are published everywhere. Chances are good that most major content sources you follow online publish their data in some type of serialized format. Heck, [even Marvel publishes an API](http://developer.marvel.com/documentation/getting_started). Look around for a "Developers" section on major websites, or ask the Google Answer-Bot.
+
+> Except for ESPN and most major sports entities :(
+  - [ESPN Dev](http://espn.go.com/apis/devcenter/)
+
+**That sounds hard. Can't you just give me a freebie?**
+
+Okay... try the [Programmable Web API Directory](http://www.programmableweb.com/apis/directory) or the [Public APIs Directory](http://www.publicapis.com/).
+
+## What Is An API Key? (5 mins)
+
+While the majority of APIs are free to use, many of them require an API "key" that identifies the developer requesting data access. This is done to regulate usage and prevent abuse. Some APIs also rate-limit developers, meaning they have caps on the free data allowed during a given time period.
+
+**Try hitting the Giphy API...**
+
+* No key: [http://api.giphy.com/v1/gifs/search?q=funny+cat](http://api.giphy.com/v1/gifs/search?q=funny+cat)
+
+* With key: [http://api.giphy.com/v1/gifs/search?q=funny+cat&api_key=dc6zaTOxFJmzC](http://api.giphy.com/v1/gifs/search?q=funny+cat&api_key=dc6zaTOxFJmzC)
+
+**It is very important that you not push your API keys to a public Github repo.**
```

Comment:
Opportunity to provide some resources (e.g., Figaro) either here or at the end of the lesson on how to deal with API key security, especially when something is being pushed to Github.